cranelift/isle/isle/src/trie_again.rs

```diff
@@ -20,6 +20,15 @@ impl BindingId {
     }
 }
 
+impl IntoIterator for TupleIndex {
+    type Item = Self;
+    type IntoIter = std::iter::Map<std::ops::Range<u8>, fn(u8) -> TupleIndex>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        (0..self.0).map(TupleIndex)
```

Comment:
This is somewhat unintuitive -- iterating over an index gives me ... all indices less than that index?

If we have to have such a method, I'd prefer for it not to be a trait impl of `IntoIterator` (this also avoids the need to specify the explicit `IntoIter` type and expose the iterator-chain innards) but rather have something like `fn up_to(self) -> impl Iterator<Item = Self> { (0..self.0).map(TupleIndex) }`.